File: thirvusoft_bpm/thirvusoft_bpm/razorpay_backup/razorpay_checkout.py
# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
# License: GNU General Public License v3. See license.txt
from __future__ import unicode_literals

import json

from six import string_types

import frappe
from frappe import _
from frappe.utils import cint, flt

no_cache = 1

expected_keys = (
	"amount",
	"title",
	"description",
	"reference_doctype",
	"reference_docname",
	"payer_name",
	"payer_email",
	"order_id",
)


def get_context(context):
	context.no_cache = 1
	# Start
	try:
		doc = frappe.get_doc("Integration Request", frappe.form_dict["token"])
		details = json.loads(doc.data)
		if(details['reference_doctype']=="Payment Request"):
			pay_doc = frappe.db.get_value('Payment Request',details['reference_docname'],'reference_doctype')
			if pay_doc == "Fees":
				fee_doc = frappe.db.get_value('Payment Request',details['reference_docname'],'reference_name')
				company = frappe.db.get_value('Fees',fee_doc,'company')
		if company:
			context.api_key = get_api_key(company)
	except:
		context.api_key = get_api_key()
	# End
	try:
		doc = frappe.get_doc("Integration Request", frappe.form_dict["token"])
		payment_details = json.loads(doc.data)

		for key in expected_keys:
			context[key] = payment_details[key]

		context["token"] = frappe.form_dict["token"]
		context["amount"] = flt(context["amount"])
		context["subscription_id"] = (
			payment_details["subscription_id"] if payment_details.get("subscription_id") else ""
		)

	except Exception as e:
		frappe.redirect_to_message(
			_("Invalid Token"),
			_("Seems token you are using is invalid!"),
			http_status_code=400,
			indicator_color="red",
		)

		frappe.local.flags.redirect_location = frappe.local.response.location
		raise frappe.Redirect


def get_api_key(company=None):
	# Start
	if company:
		api_key = frappe.db.get_value("Company", company, "api_key")
	else:
		api_key = frappe.db.get_value("Razorpay Settings", None, "api_key")
	# End
	if cint(frappe.form_dict.get("use_sandbox")):
		api_key = frappe.conf.sandbox_api_key

	return api_key


@frappe.whitelist(allow_guest=True)
def make_payment(razorpay_payment_id, options, reference_doctype, reference_docname, token):
	data = {}

	if isinstance(options, string_types):
		data = json.loads(options)

	data.update(
		{
			"razorpay_payment_id": razorpay_payment_id,
			"reference_docname": reference_docname,
			"reference_doctype": reference_doctype,
			"token": token,
		}
	)

	data = frappe.get_doc("Razorpay Settings").create_request(data)
	frappe.db.commit()
	return data

File: thirvusoft_bpm/thirvusoft_bpm/razorpay_backup/razorpay_checkout.js
$(document).ready(function(){
	(function(e){
		var options = {
			"key": "{{ api_key }}",
			"amount": cint({{ amount }} * 100), // 2000 paise = INR 20
			"name": "{{ title }}",
			"description": "{{ description }}",
			"subscription_id": "{{ subscription_id }}",
			"handler": function (response){
				razorpay.make_payment_log(response, options, "{{ reference_doctype }}", "{{ reference_docname }}", "{{ token }}");
			},
			"prefill": {
				"name": "{{ payer_name }}",
				"email": "{{ payer_email }}",
				"order_id": "{{ order_id }}"
			},
			"notes": {{ frappe.form_dict|json }}
		};

		var rzp = new Razorpay(options);

		frappe.call({
			'method':"thirvusoft_bpm.thirvusoft_bpm.custom.py.razorpay_checkout.check_expiry_date",
			'args':{
				'token':'{{token}}'
			},
			callback:function(res){
				if(res.message==true)
				{
					rzp.open();
				}
				else{
					frappe.throw({
						title: __("Remainder"),
						message: __(res.message[0]),
					});
				}
			}
		})

	
		//	e.preventDefault();
	})();
})

frappe.provide('razorpay');

razorpay.make_payment_log = function(response, options, doctype, docname, token){
	$('.razorpay-loading').addClass('hidden');
	$('.razorpay-confirming').removeClass('hidden');

	frappe.call({
		method:"frappe.templates.pages.integrations.razorpay_checkout.make_payment",
		freeze:true,
		headers: {"X-Requested-With": "XMLHttpRequest"},
		args: {
			"razorpay_payment_id": response.razorpay_payment_id,
			"options": options,
			"reference_doctype": doctype,
			"reference_docname": docname,
			"token": token
		},
		callback: function(r){
			if (r.message && r.message.status == 200) {
				window.location.href = r.message.redirect_to
			}
			else if (r.message && ([401,400,500].indexOf(r.message.status) > -1)) {
				window.location.href = r.message.redirect_to
			}
		}
	})
}
